### DISCUSSION TITLE
Checking questions only if it's answered

### DISCUSSION THREAD

**Discussion by kushshiv** [IndianOpenSourceFoundation/dynamic-cli#133] [General]
Is there any specific reason for checking the question only if it has been answered. Some solutions in stackoverflow are not marked answered but are helpful.

    def get_que(self, json_data):
        """
        This function returns the list of ids of the questions
        that have been answered, from the response that we get
        from the make_request function.
        """
        que_id = []
        for data in json_data['items']:
            if data["is_answered"]:
                que_id.append(data["question_id"])
        return que_id

**Comment by lainq** [comment]
Well, the command is expected to fetch answers of a specific question from stackoverflow. I don't think there's any use of displaying empty answers, What do you think 🤔 ?

**Reply by GouravSardana** [reply]
Yep. It doesn't make sense to fetch the question which are not answered.

**Reply by GouravSardana** [reply]
If @kushshiv thinks in any different way, could you please explain?

**Reply by kushshiv** [reply]
Some questions has a solution, and people found its working for them, and even has lot of upvotes. But it has not been marked accepted(Green Tick).  Like this one below :
https://stackoverflow.com/questions/29092970/importerror-cannot-import-name-webdriver?rq=1


It could be better if we can check `answer_count` rather than `is_answered` and show those questions as well irrespective of whether it's marked accepted or not. What are your views...??

```
{
   "items":[
      {
        {
           "tags":[
              "nginx"
           ],
           "owner":{
              "reputation":3135,
              "user_id":4611304,
              "user_type":"registered",
              "accept_rate":52,
              "profile_image":"https://www.gravatar.com/avatar/a6c4b8dc9b9a765522cdfed07cbb1c92?s=256&d=identicon&r=PG&f=1",
              "display_name":"techguy2000",
              "link":"https://stackoverflow.com/users/4611304/techguy2000"
           },
           "is_answered":false,
           "view_count":57,
           "answer_count":3,
           "score":0,
           "last_activity_date":1491544705,
           "creation_date":1491428153,
           "question_id":43242052,
           "content_license":"CC BY-SA 3.0",
           "link":"https://stackoverflow.com/questions/43242052/what-is-the-syntax-for-a-nginx-url-rewrite",
           "title":"What is the syntax for a nginx url rewrite"
        }
   ],
   "has_more":false,
   "quota_max":300,
   "quota_remaining":222
}
```

**Reply by lainq** [reply]
That makes sense. If that's the case, I think we should change it to  `answer_count` .  I thought `is_answered` meant if the anyone answered the question.

**Reply by kushshiv** [reply]
Just checked, to be more clear,
`is_answered` will be True if :

- The question has an accepted answer
- The question has an answer with a score > 0

So if question is answered and no one upvoted, then can we show the questions...?? I think we can still show them. As most of the new questions does not have upvotes.

**Reply by GouravSardana** [reply]
Yep that makes sense. Go ahead
